H/HDF5/build_tarballs.jl

```diff
@@ -45,46 +50,397 @@ if [[ ${target} == *mingw* ]]; then
+
+mkdir build
+pushd build
+
+cmake \
+    -DCMAKE_FIND_ROOT_PATH=${prefix} \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DBUILD_STATIC_LIBS=OFF \
+    -DBUILD_TESTING=OFF \
+    -DHDF5_BUILD_EXAMPLES=OFF \
+    -DHDF5_BUILD_HL_LIB=ON \
+    -DHDF5_BUILD_TOOLS=ON \
+    -DHAVE_IOEO_EXITCODE=0 \
+    -DTEST_LFS_WORKS_RUN=0 \
+    -DH5_LDOUBLE_TO_LONG_SPECIAL_RUN=1 \
+    -DH5_LDOUBLE_TO_LONG_SPECIAL_RUN__TRYRUN_OUTPUT= \
+    -DH5_LONG_TO_LDOUBLE_SPECIAL_RUN=1 \
+    -DH5_LONG_TO_LDOUBLE_SPECIAL_RUN__TRYRUN_OUTPUT= \
+    -DH5_LDOUBLE_TO_LLONG_ACCURATE_RUN=0 \
+    -DH5_LDOUBLE_TO_LLONG_ACCURATE_RUN__TRYRUN_OUTPUT= \
+    -DH5_LLONG_TO_LDOUBLE_CORRECT_RUN=0 \
+    -DH5_LLONG_TO_LDOUBLE_CORRECT_RUN__TRYRUN_OUTPUT= \
+    -DH5_DISABLE_SOME_LDOUBLE_CONV_RUN=1 \
+    -DH5_DISABLE_SOME_LDOUBLE_CONV_RUN__TRYRUN_OUTPUT= \
+    ..
+cmake --build . --config RelWithDebInfo --parallel ${nproc}
+cmake --build . --config RelWithDebInfo --parallel ${nproc} --target install
+
+popd
+
 else
-    cd ${target}
-
-    # Delete the C++ and Fortran libraries, we can't use them because they'd restrict the
-    # ABIs we can use this package for (e.g., *only* libgfortran5 and C++11 string ABI),
-    # which would be overly restrictive, as Julia on Linux comes with libgfortran4, thus
-    # making this package completely unusable there.  The only way to use the C++ and
-    # Fortran libraries is for BinaryBuilder to be able to actually build them:
-    # <https://github.com/JuliaPackaging/Yggdrasil/issues/4760>.
-    rm -f lib/{*_cpp*,*_fortran*}
-    mv -v lib/* ${libdir}
-    mv -v include/* ${includedir}
 
-    install_license info/licenses/COPYING
+# Option 2: Build with configure
+# This is the currently supported way.
+
+# Patch `configure.ac`:
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/configure.ac.patch
+
+# Prepare the files `H5init.c` and `config.saved` that contain predetermined
+# configuration information
+mkdir saved
+case "${target}" in
+    aarch64-apple-darwin*)
+        cp ../files/darwin-arm64v8/* saved
+        ;;
+    aarch64-linux-*)
+        cp ../files/debian-arm64v8/* saved
+        ;;
+    arm-linux-*)
+        cp ../files/debian-arm32v7/* saved
+        ;;
+    i686-linux-*)
+        cp ../files/debian-i386/* saved
+        ;;
+    i686-w64-mingw32)
+        # sizeof(long double) == 12
+        # layout seems to be 16-bit sign+exponent and 64-bit mantissa
+        # same as for Linux
+        cp ../files/msys2-mingw32/* saved
+        ;;
+    powerpc64le-linux-*)
+        cp ../files/debian-ppc64le/* saved
+        ;;
+    x86_64-apple-darwin*)
+        cp ../files/darwin-amd64/* saved
+        ;;
+    x86_64-linux-*)
+        cp ../files/debian-amd64/* saved
+        ;;
+    x86_64-*-freebsd*)
+        # same as for Linux
+        cp ../files/freebsd-amd64/* saved
+        ;;
+    x86_64-w64-mingw32)
+        # sizeof(long double) == 16
```

Comment:
Is this actually 16 bytes? Or is it 10 bytes aligned to 16 bytes?

Reply:
I don't think alignment can be larger than size.

The above says the size is 16 bytes, likely 10 bytes of data with 6 bytes of padding.

Reply:
ah, you're right: compilers report `sizeof` to include padding.

Reply:
I can prove it in your local friendly binary builder environment with a `static_assert`.

```
sandbox:${WORKSPACE}/srcdir # cat static_test.cpp 
#include<iostream>

int main() {
    static_assert(sizeof(long double) != 16);
}

sandbox:${WORKSPACE}/srcdir # x86_64-w64-mingw32-g++ static_test.cpp 
static_test.cpp: In function ‘int main()’:
static_test.cpp:4:5: error: static assertion failed
     static_assert(sizeof(long double) != 16);
```